python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -0,0 +1,491 @@
+# Copyright (c) 2021 NVIDIA CORPORATION
+
+import numbers
+import numpy as np
+
+from shapely.geometry import (
+    Point,
+    MultiPoint,
+    LineString,
+    MultiLineString,
+    Polygon,
+    MultiPolygon,
+)
+from typing import TypeVar, Union
+
+import cudf
+from cudf.core.column import ColumnBase, NumericalColumn
+
+from cuspatial.geometry.geoarrowbuffers import GeoArrowBuffers
+
+
+T = TypeVar("T", bound="GeoColumn")
+
+
+class GeoMeta:
+    """
+    Creates  input_types and input_lengths for GeoColumns that are created
+    using native GeoArrowBuffers. These will be used to convert to GeoPandas
+    GeoSeries if necessary.
+    """
+
+    def __init__(self, meta: Union[GeoArrowBuffers, dict]):
+        if isinstance(meta, dict):
+            self.input_types = meta["input_types"]
+            self.input_lengths = meta["input_lengths"]
+            return
+        buffers = meta
+        self.input_types = []
+        self.input_lengths = []
+        if hasattr(buffers, "points"):
+            self.input_types += list(np.repeat("p", len(buffers.points)))
+            self.input_lengths += list(np.repeat(1, len(buffers.points)))
+        if hasattr(buffers, "multipoints"):
+            self.input_types += list(np.repeat("mp", len(buffers.multipoints)))
+            self.input_lengths += list(np.repeat(1, len(buffers.multipoints)))
+        if hasattr(buffers, "lines"):
+            if len(buffers.lines.mlines) > 0:
+                self.input_types += list(
```

Comment:
Could use `itertools.repeat()` here instead of a numpy array. Also, stylistically `x.extend(y)` may be preferable over `x += list(y)`.

Reply:
Ditto below in this function.

Reply:
done